src/engine/renderer/tr_image.cpp

```diff
@@ -1758,12 +1758,21 @@ static void R_LoadImage( const char **buffer, byte **pic, int *width, int *heigh
 		}
 	}
 
-	// the file isn't there, try again without the extension
-	COM_StripExtension3( token, filename, MAX_QPATH );
-
+	// if the file isn't there, maybe the file path did not have any extension
+	// but the file has a dot in its name that was mistakenly taken as an extension
 	const char *prefix;
 	int bestLoader = R_FindImageLoader( filename, &prefix );
 
+	if ( bestLoader == -1 )
+	{
+		// if the file isn't there, maybe its filename does not have extension
+		// or there is no file with such extension
+		// or there is no codec available for this file format
+		COM_StripExtension3( token, filename, MAX_QPATH );
```

Comment:
Prefer sizeof(filename) instead of hard-coding the buffer size

Reply:
This code already existed before, it's just moved to a new place.
Isn't it better to postpone such change to another PR?

To be honest I first started to rewrite the code to use std::string like the SoundCodec does, but then I dropped my work in progress because it would not be related to the initial purpose of the PR.